### PR TITLE
hplip: update to 3.22.6.

### PR DESCRIPTION
**Pull request by dataCobra**
* Remove 1 patch and add 1 build dependency
* Tested both packages with hplip-scan & xsane on HP Neverstop Laser MFP 120x.
* Continue from PR hplip: update to 3.22.6 #37287

I've split the former PR into two separate branches to make sure at least hplip gets updated.

The corresponding `hplip-plugin` PR is #39147.

#### Testing the changes
- I tested the changes in this PR: **YES - tested by @biopsin and myself**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl (crossbuild)
  - armv7l (crossbuild)
  - armv6l-musl (crossbuild)

### CODE REVIEW

**Comment by dataCobra** [comment]
Hey @paper42,

I use this now for several weeks without an issue.

I think it can and should be merged.

**Comment by dataCobra** [comment]
Added myself as the maintainer for the package.

I've also corrected the commit message.

File: srcpkgs/hplip/template [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Template file for 'hplip'
+# Must be in sync with hplip-plugin package
```
Comment (ahesford) [inline srcpkgs/hplip/template:2]: Drop this comment. The hplip-plugin package doesn't exist. Even if it is accepted, we shouldn't demand that people keep a non-free, restricted package in sync. If HP delays or abandons releases of the plug-in package, I'd rather people update this regularly than wait on a package fee people are likely to discover and use anyway.

If the versions really must be tied, the right thing to do is add a version-restricted dependency in the hplip-plugin package.

File: srcpkgs/hplip/template [hunk 2]
```diff
@@ -31,12 +32,12 @@ makedepends="openssl-devel python3-devel libxml2-python3 cups-devel sane-devel
 depends="python3-gobject python3-dbus desktop-file-utils
  foomatic-db foomatic-db-engine python3-distro"
 short_desc="HP Linux Imaging and Printing"
-maintainer="Orphaned <orphan@voidlinux.org>"
+maintainer="dataCobra <datacobra@thinkbot.de>"
```
Comment (ahesford) [inline srcpkgs/hplip/template:35]: Don't adopt yet. We want to see a longer contribution history than one month, and at least a few cycles of updates to this package, before putting somebody's name on it.

**Review by ahesford — CHANGES_REQUESTED** [review]
There's no need to ping paper, who has already disavowed familiarity with hplip.

There's no need to ping anybody, really, because we all track the PR list when we have time and interest.

File: srcpkgs/hplip/template (hunk 2 above)
Comment (dataCobra) [reply srcpkgs/hplip/template:35]: This is new to me... I'm already the maintainer for mercurial for which I've only created two PRs.

But I'll undo this change then.

File: srcpkgs/hplip/template (hunk 1 above)
Comment (dataCobra) [reply srcpkgs/hplip/template:2]: I'll remove it.

**Pull request merged**